[3.0.5] 投稿カテゴリの設定が適用されない問題を修正

nkpoid opened the pull request:
## WHAT 
- same as title
- 多少のリファクタリングを行った(記法の統一)

## バグ内容と修正箇所
カスタム投稿タイプではカテゴリが存在しないため、通常投稿のみカテゴリはチェックすれば良い。
また、設定でチェックボックスを外して保存した場合、`'false'`が保存される場合と、`''`が保存される場合が混在しているため、両方のパターンに対応するようにした。(今回のバグの直接の原因はここ)

nkpoid committed:
投稿カテゴリの設定が適用されない問題を修正

nkpoid committed:
add changelog

a-r-g-v commented on classes/push7-post.php:
```diff
@@ -65,13 +65,12 @@ public function push($post, $is_rp=false) {
-      return;
-    }
+    $post_type = get_post_type($post_id);
+    $post_id = $this->get_post_id($post);
```
なぜ `$post_id` の再取得をしているのでしょうか？

nkpoid replied:
これは頭ついてませんでした

a-r-g-v commented on classes/push7-post.php:
```diff
@@ -65,13 +65,12 @@ public function push($post, $is_rp=false) {
-      return;
-    }
+    if ($this->check_ignored_posttype($post_id)) return;
+    // 通常投稿の場合のみカテゴリチェックを行う
```
このコメントはなくて良いと思います。下の文と言っていることが同じだと思うからです。
（言及するべきは、 「なぜ通常投稿の場合のみカテゴリチェックを行うのか？」だと思います）

a-r-g-v commented on setting.php:
```diff
@@ -120,12 +120,11 @@ function show_advances_info() {
                   <label for="<?= $name; ?>">
                     <input type="checkbox" name="<?= $name;?>" value="true" <?php checked("true", get_option($name)) ?>>
-                    <?= $post_type == 'post' ? 'post(通常の投稿)' : $post_type; ?>
+                    <?= $post_type === 'post' ? 'post(通常の投稿)' : $post_type; ?>
```
`==` から `===` に変更してもよいのでしょうか？

nkpoid replied:
はい

a-r-g-v commented on classes/push7-post.php:
```diff
@@ -65,13 +65,12 @@ public function push($post, $is_rp=false) {
-    }
+    if ($this->check_ignored_posttype($post_id)) return;
+    // 通常投稿の場合のみカテゴリチェックを行う
+    if ($post_type === 'post' && $this->check_ignored_category($post_id)) return;
```
`==` から`===` に変更してもよいのでしょうか？

nkpoid committed:
reflect review

a-r-g-v reviewed (changes requested):
CIの修正をお願いします（コードはLGTMです）

nkpoid committed:
ci修正
> You’ll need to specify the full version for now.
https://travis-ci.community/t/running-phpenv-runs-rbenv-instead/1851

nkpoid committed:
try fix
